src/threadNames.cpp

```diff
@@ -0,0 +1,48 @@
+#include "threadNames.h"
+#include "profiler.h"
+
+ThreadNames ThreadNames::_instance;
+
+void ThreadNames::update(jthread thread) {
+    MutexLocker ml(_lock);
```

Comment:
Mutex is only needed to protect `_storage`; there is no need to perform other possibly heavy operations like `GetThreadInfo` under lock. I'd suggest to keep critical section as short as possible.

Reply:
OK.
The idea was to use one lock for storage and for `_eetop` updates.